dqc_us_rules/dqc_us_0018.py

```diff
@@ -262,20 +259,22 @@ def _fact_checkable(fact):
     )
 
 
-def _catch_linkbase_deprecated_errors(val):
+def _catch_linkbase_deprecated_errors(val, deprecated_concepts):
     """
     Check for unused concept relationships of standard taxonomy elements
     and catches abstract deprecated concepts in linkbases
 
     :param val: :class: '~arelle.ValdiateXbrl.ValidateXbrl'
     :return: Returns deprecated concepts from the linkbase
-    :rtype: :class: '~arelle.ModelInstanceObject.ModelConcept'
+    :rtype: None
     """
     relationships = val.modelXbrl.relationshipSet(XbrlConst.parentChild)
     for rel in relationships.modelRelationships:
         for concept in (rel.fromModelObject, rel.toModelObject):
             if _deprecated_concept(val, concept):
-                yield concept
+                if not deprecated_concepts.get(concept.name):
```

Comment:
there's a concept.name attribute? I thought you had to grab the qname localname?